File: bmodle/src/main/java/com/hong_world/bmodle/modle/TasksDataSource.java
package com.hong_world.bmodle.modle;

import com.hong_world.common.bean.FeedArticleListData;

import io.reactivex.Observable;

/**
 * Date: 2018/8/13. 10:45
 * Author: hong_world
 * Description:
 * Version:
 */
public interface TasksDataSource {

    interface GetTaskCallback<C> {

        void onTaskLoaded(C bean);

        void onDataNotAvailable(String type, String msg);
    }
    Observable<FeedArticleListData> getFeedArticleList(int num);
}

File: kotlin-module/src/main/java/com/hong_world/kotlin_module/modle/TasksDataSource.kt
package com.hong_world.kotlin_module.modle

import com.hong_world.kotlin_module.bean.FeedArticleListData
import io.reactivex.Observable

/**
 * Date: 2018/9/25. 16:21
 * Author: hong_world
 * Description:
 * Version:
 */
interface TasksDataSource {
    interface GetTaskCallback<C> {

        fun onTaskLoaded(bean: C)

        fun onDataNotAvailable(type: String, msg: String)
    }

    abstract fun getFeedArticleList(num: Int): Observable<FeedArticleListData>?
}